libsplinter/src/auth/oauth/rest_api/mod.rs

```diff
@@ -28,6 +28,12 @@ pub trait AuthUserOperation: Sync + Send {
     /// Executes a save operation on the given user info.
     fn save_user_info(&self, user_info: &UserInfo) -> Result<(), InternalError>;
 
+    /// Executes a `RemoveUserAccessOperation` on the user with the given access token.
+    ///
+    /// Returns an optional user identity, signifying the user was found in underlying storage
+    /// and successfully updated.
+    fn remove_user_access(&self, access_token: &str) -> Result<Option<String>, InternalError>;
```

Comment:
Why does this return the user's identity? Is it used by anything? Seems like a bool would be more appropriate. I'd also rename this to `remove_user_access_token` to be more clear.

Reply:
Does it actually need to return anything? I.e. could it be `Result<(), InternalError>`?

Reply:
The issue I was running into, is the function to fetch the `OAuthUser` returns `Ok(None)` if it doesn't find the user. I assumed we would want to return a 401 in this case. So, since the `identity` field will always be there, if it is returned as `None`, that means the user doesn't exist and the `/oauth/logout` endpoint can return a 401.

Reply:
Updated how the access token is handled, as suggested in another comment, so this method now returns just `Result<(), InternalError>`.